### PR TITLE
Add pipelines as code Repository form

### PR DESCRIPTION
**Pull request by karthikjeeyar**
**Fixes**: 

https://issues.redhat.com/browse/ODC-6461
**Analysis / Root cause**: 

User can add a Git repository to pipeline as code

**Solution Description**: 

1. Supports Github, Gitlab and bitbucket webhook method
2. Shows the sample pipelinerun templates based on the runtime if it is not available then fallback to default template.
3. Added Tech Preview badge.

**Screen shots / Gifs for design review**: 

image

image

image

**Unit test coverage report**: 

```
createRepositoryResources
    ✓ creates only repository CR when no token is provided (4ms)
    ✓ creates repository CR and secret when token is provided (1ms)
    ✓ creates repository CR with custom host when repository host is not GitHub (1ms)
  detectGitType
    ✓ should return the correct git provider (1ms)
    ✓ should return git provider as unsure type if it is self hosted
    ✓ should return  invalid type
  createRepositoryName
    ✓ should create repository name with prefix (1ms)
    ✓ should convert and return a valid repo name (1ms)
  recommendRepositoryName
    ✓ should return undefined for invalid values
    ✓ should recommend a repo cr name based on the url (1ms)
  getPipelineRunDefaultTemplate
    ✓ should return default fallback template if a custom template is not available
    ✓ should contain default name if repo name is invalid
    ✓ should return custom template (1ms)
  getPipelineRunTemplate
    ✓ should return a default template if custom template is not available
    ✓ should return custom template
```
**Test setup:**

1. Install 1.8 OSP operator [url](https://artifacts.ospqa.com/builds/1.8.0/ )(use vpn)
2. Setup Github App from Admin-> pipelines
3. While creating Repository, Install the github app in your repository.
4. Follow steps on repository overview page to add the .tekton directory in your repository.
5. Create a push/pull request, a pipelinerun will be triggered in the target namespace.

Note:
Some fields will not be represented in the UI when OSP 1.7.x is installed 

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge

### CODE REVIEW

**Comment by MariaLeonova** [comment]
This looks great, Karthik!
One small thing I noticed - option for Git access token secret field is missing a description. It should say: "Secret with the Git access token for pulling pipeline and tasks from your Git repository", and be shown once user selects this option.

File: frontend/packages/pipelines-plugin/src/components/repository/form-fields/CopyPipelineRunButton.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    ? t('pipelines-plugin~Copied to clipboard')
+    : t('pipelines-plugin~Copy to clipboard');
+  const tooltipContent = [
+    <span className="co-nowrap" key="nowrap">
+      {tooltipText}
+    </span>,
+  ];
+
+  return (
+    <div data-test={'pipelinerun-copy-btn'}>
```
Comment (vikram-raj) [inline frontend/packages/pipelines-plugin/src/components/repository/form-fields/CopyPipelineRunButton.tsx:21]:
```suggestion
    <div data-test="pipelinerun-copy-btn">
```

File: frontend/packages/pipelines-plugin/src/components/repository/sections/WebhookSection.tsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    }
+
+    return <div data-test={`${values.gitProvider}-helptext`}>{helpText}</div>;
+  };
+
+  return (
+    <FormSection fullWidth extraMargin>
+      <RadioGroupField
+        name="webhook.method"
+        label="Secret"
```
Comment (vikram-raj) [inline frontend/packages/pipelines-plugin/src/components/repository/sections/WebhookSection.tsx:123]: Add translation
```suggestion
        label=t('pipelines-plugin~Secret')
```

**Review by vikram-raj — COMMENTED** [review]
Thanks @karthikjeeyar. Verified the changes and everything works as expected.

/lgtm
/label qe-approved

**Comment by karthikjeeyar** [comment]
Thanks @MariaLeonova . I have updated it
image
cc: @vikram-raj

**Comment by karthikjeeyar** [comment]
Propagating Acks from the [Epic](https://issues.redhat.com/browse/ODC-6461)

/label docs-approved
/label px-approved

**Comment by MariaLeonova** [comment]
Thank you for the update, @karthikjeeyar ! Looks great.
Regarding the usability - would be great to have a quick test with a user, if you have someone in mind.

**Pull request merged**